### DISCUSSION TITLE
请问如何将SDK接入Unity呢？

### DISCUSSION THREAD

**Discussion by Sherlock-White** [Tencent/libpag#1638] [Q&A] [answered]
我现在通过文档https://pag.art/docs/sdk.html 里面提到的Windows接入指南，成功运行demo，也成功执行到PAG【Angle backend】中第二步编译生成pag.lib和pag.dll
但我非常不明确要继续将SDK接入Unity该怎么办，想问问有大致的思路吗？

**Comment by kevingpqi123** [comment, marked as answer]
接入层面可能很快，pag 的渲染需要外部提供一个 OpenGL 的渲染环境，简单做下适配就可以了渲染出内容
比较难的是后续的性能优化，这是需要投入大量时间去解决的

**Reply by SilenSang** [reply]
我在 Unity native plugin 的渲染回调中尝试调用 `PAGSurface::MakeFrom(const BackendTexture& texture, ...);` 是失败的，原因可能是 libpag 无法使用 Unity 所提供的 OpenGL context。我已将 Unity 的 Graphics API for Windows 由默认的 DX 11 改为 OpenGL ES 2.0。请问您说的“适配”应该如何做？

**Reply by kevingpqi123** [reply]
这里不仅仅是使用层面的修改，还会涉及到源码层面的修改适配，需要对 libpag 的平台层封装很熟悉，建议先系统了解下 libpag 的源码，且适配完还要处理性能优化问题，OpenGL 的渲染逻辑和 Unity 的渲染逻辑还不太一样，这里的工作量有点大，不是三两句话可以说明白的

**Reply by SilenSang** [reply]
OK, thanks.
